core/Plugin/Dimension/VisitDimension.php

```diff
@@ -346,4 +347,45 @@ public static function getDimensions(Plugin $plugin)
 
         return $instances;
     }
+
+    /**
+     * Sort a key => value array descending by the number of occurances of the key in the supplied table and column
+     *
+     * @param array     $array              Key value array
+     * @param DataTable $table              Datatable from which to count occurances
+     * @param string    $keyColumn          Column in the datatable to match against the array key
+     * @param int       $maxValuesToReturn  Limit the return array to this number of elements
+     *
+     * @return array    An array of values from the source array sorted by most occurances, descending
+     */
+    public function sortStaticListByUsage(array $array, DataTable $table, string $keyColumn, int $maxValuesToReturn) : array
+    {
+        // Convert to multi-dimensional array and count the number of visits for each browser name
+        foreach ($array as $k => $v) {
+            $array[$k] = ['count' => 0, 'name' => $v];
+        }
+        foreach ($table->getRows() as $row) {
+            if (isset($row[$keyColumn]) && $row[$keyColumn] !== 'xx') {
+                $array[$row[$keyColumn]]['count']++;
+            }
+        }
+        // Sort by most visits descending
+        uasort($array, function($a, $b) {
+            return $a <=> $b;
+        });
+        $array = array_reverse($array, true);
+
+        // Flatten and limit the return array
+        $flat = [];
+        $i = 0;
+        foreach ($array as $k => $v) {
+            $flat[$k] = $v['name'];
+            $i++;
+            if ($i == ($maxValuesToReturn - 1)) {
+                break;
+            }
+        }
+        $flat[] = 'Unknown';
```

Comment:
Does it make sense to have `Unknown` always in the list? Even if it might not have visits?

Reply:
The previous code appended `Unknown` to the bottom of the list, but you are right, that no longer makes sense. I've changed this to add `Unknown` the list from the beginning and any unmatched row values should now be counted as `Unknown`. So `Unknown` should only be returned in the suggested values list if it has enough visits.